client/wsclient.go

```diff
@@ -131,8 +131,28 @@ func (c *wsClient) tryConnectOnce(ctx context.Context) (err error, retryAfter sh
 			c.common.Callbacks.OnConnectFailed(ctx, err)
 		}
 		if resp != nil {
-			c.common.Logger.Errorf(ctx, "Server responded with status=%v", resp.Status)
 			duration := sharedinternal.ExtractRetryAfterHeader(resp)
+			if resp.StatusCode >= 300 && resp.StatusCode < 400 {
+				// very liberal handling of 3xx that largely ignores HTTP semantics
+				redirect, err := resp.Location()
+				if err != nil {
+					c.common.Callbacks.OnConnectFailed(ctx, err)
```

Comment:
I am not sure we need this second call to OnConnectFailed. The callback was already called on line 131. This would be the second call per connection attempt and I don't think that's a good thing to do.
Unclear what extra value does this call give to the user. They could log the error, but we already do that here.

Reply:
That's a fair callout. In this case it was just propping up the test, as I didn't have another way of easily inspecting the error condition here: https://github.com/open-telemetry/opamp-go/pull/251/files#diff-9adf0e52d7c19860aa5771807f97b10d8ab7d5acc71acc2fa1c7f322bc4e46c6R234

I don't think the tests should push the implementation around, but I also am not sure how better to deal with this in test. Any suggestions?

Reply:
One possible way is to have an unexported field in wsClient that we set here (e.g. `lastRespErr` or similar) and use it in the test.